### ISSUE TITLE
SSH key forwarding not working

### ISSUE THREAD

**Issue by zackphilipps** [Varying-Vagrant-Vagrants/VVV#1720]
## Expected Behavior
Ability to clone private repos hosted on Bitbucket.

## Current Behavior
I get this when I run `vagrant ssh` and attempt to clone.
```
repository access denied. deployment key is not associated with the requested repository.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

## Possible Solution
Not sure. This could very well be an incredible PEBKAC.

## Steps to Reproduce (for bugs)

1. Download latest versions of Vagrant, VirtualBox, and VVV.
2. `vagrant ssh`
3. `git clone git@bitbucket.org:<repo>.git`

## Context
I'm trying to create my own site template. I see that this issue has come up in the past (#1225), but I've had no luck with any of the suggestions I've found online. `ssh -T git@bitbucket.org` also lists a bunch of repos that I don't recognize at all and says I have read access to them.

## Your Environment

* VVV version: 2.5.2
* VVV Git Branch: develop
* Vagrant version: 2.2.3
* VM Provider name: VirtualBox
* VM Provider version: 5.2.24r128163
* Operating System and version: macOS Mojave 10.14.2

**Comment by tomjn** [comment]
@zackphilipps  can you confirm that you can use `git clone` on the host machine?

**Comment by zackphilipps** [comment]
@tomjn yep, I use SSH exclusively for git remotes. Here's what I get:

```bash
vagrant@vvv:~$ ssh -T git@bitbucket.org
authenticated via a deploy key.


You can use git or hg to connect to Bitbucket. Shell access is disabled.

This deploy key has read access to the following repositories:
```

It appears the username is missing???

**Comment by tomjn** [comment]
Some more things:

 - Is this new or has this always been the case?
 - When you say that's what you get, that's inside the VM, what about outside the VM on the host?
 - Any mentions of deploy key are from bitbuckets end, not VVV
 - Are you absolutely sure you have an SSH Agent configured correctly? And that your main SSH key is indeed your main SSH key? Wherever this deploy key is coming from, it's not something I've seen before, and it's not something VVV is bundling

In the guest VM ( VVV ):

```
vagrant@vvv:~$ ssh -T git@bitbucket.org
Warning: Permanently added the RSA host key for IP address '18.205.93.2' to the list of known hosts.
logged in as tomjn.

You can use git or hg to connect to Bitbucket. Shell access is disabled.
vagrant@vvv:~$ 
```

On my host OS ( MacOS ) I get the same:

```
~/dev/www/vvv git::develop 
❯ ssh -T git@bitbucket.org
logged in as tomjn.

You can use git or hg to connect to Bitbucket. Shell access is disabled.
```

Right now it's ambiguous what your current status is for all possible configurations, except running the command inside VVV:

 - There's no clearly labelled data for what happens with the host system
 - If this is new behaviour, or if it has never worked, or did it fail on an update of a piece of software?
 - If this is an SSH forwarding problem or a bitbucket problem
 - You mention various repositories but censor them, what are they? Who do they belong to?

So I present the following:

|                            	| VVV 	| host OS 	|
|----------------------------	|-----	|---------	|
| `ssh -T git@bitbucket.org` 	| 🚫   	|         	|
| `ssh -T git@github.com`    	|     	|         	|

So that you don't need to faff around with markdown table generators:

```
|                            	| VVV 	| host OS 	|
|----------------------------	|-----	|---------	|
| `ssh -T git@bitbucket.org` 	| 🚫   	|         	|
| `ssh -T git@github.com`    	|     	|         	|
```

**Comment by tomjn** [comment]
Also `ssh-add -l` will list the ssh keys added by their fingerprints, but do this on **both** the guest VM  and the host OS

**Comment by tomjn** [comment]
Also this doc seems to have a lot more useful commands than I knew about:

https://developer.github.com/v3/guides/using-ssh-agent-forwarding/

In particular:

```shell
echo "$SSH_AUTH_SOCK"
ssh -v example.com
ssh-add -L
```

and some other checks on config files

**Comment by zackphilipps** [comment]
**TL;DR:** In a bizarre turn of events, `ssh-add -l` listed two keys. I ran `ssh-add -D` to remove all identities and then `ssh-add -k` again and only had one. Now everything works as expected. Thanks for your help!

**More details if you care:**

I have no idea where this other key came from; it wasn't mine. I don't use GitLab, but when I ran `ssh -T git@gitlab.com` I got this:

```bash
$ ssh -T git@gitlab.com
The authenticity of host 'gitlab.com (35.231.145.151)' can't be established.
ECDSA key fingerprint is SHA256:HbW3g8zUjNSksFbqTiUWPWg2Bq1x8xdGUrliXFzSnUw.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'gitlab.com,35.231.145.151' (ECDSA) to the list of known hosts.
Welcome to GitLab, @Copperis!
```

@Copperis? That ain't me!

Yesterday, I made a LOT of changes troubleshooting this. So it's hard to tell if the rogue key was introduced yesterday or sometime before. I could've sworn that `ssh-add -L` listed one key yesterday, but the output is less discernible than `ssh-add -l` (lowercase), so I could've just overlooked the second key.

At one point I had a Bitbucket key generated by SourceTree in my `.ssh` folder and then I removed it. So maybe `ssh-add -k` added that one as well _at one point_? Then again, I have another entry for GitHub (I'm assuming this was generated by the GitHub desktop app _at one point_) that I have not removed, yet `ssh-add -k` currently only adds my main key.

Maybe installing OpenSSH with Homebrew introduced some quirk as well.

🤷‍♂️

**Comment by tomjn** [comment]
You may want to let Copperis know! Glad you got it sorted

**Comment by zackphilipps** [comment]
@Copperis Not just that, but I would've had to have the _key pair_, right? To successfully auth? That's pretty crazy.

I was able to retrieve the SourceTree-generated key pair from my Trash and attempt `ssh-add -k` again. It only added my main key. So that seemingly rules that out. Not that it ever seemed likely. I guess I could send you that public key via email and see if it matches one you have.

As far as I can tell, the key pair had to have been installed as part of a package. Have you published or contributed to any Homebrew packages? Or Vagrant or VirtualBox? Or anything even remotely related to ssh-agent? As it happens, I updated OpenSSH yesterday.

I also use a ton of other package managers. 😖 

🤔 🤔 🤔

**Comment by Copperis** [comment]
@zackphilipps Would be great if you could send me the public key! I added my email to my profile.
Though I think it will not match anything.

I think the Vagrant insecure keypair got added to my account somehow.

**Comment by zackphilipps** [comment]
@Copperis I sent the public key, but I think you are right. When I ran `ssh -T git@bitbucket.org` from within vagrant, I got a list of repos I had read access to. Next to each repo, it said something like "vagrant insecure key."

Hope that helps!
